apps/web/lib/auth-edge/index.ts

```diff
@@ -0,0 +1,91 @@
+import { NextRequest } from "next/server";
+import { getSearchParams } from "@dub/utils";
+import { Session, hashToken } from "@/lib/auth";
+import { ratelimit } from "@/lib/upstash";
+import { getUserFromApiKeyViaEdge, updateApiKeyViaEdge } from "../planetscale";
+
+interface WithAuthHandler {
+  ({
+    req,
+    params,
+    searchParams,
+    headers,
+    session,
+  }: {
+    req: NextRequest;
+    params: Record<string, string>;
+    searchParams: Record<string, string>;
+    headers?: Record<string, string>;
+    session: Session;
+  }): Promise<Response>;
+}
+
+export const withAuthEdge =
+  (handler: WithAuthHandler, {}) =>
```

Comment:
The empty object {} in the `withAuthEdge` function. What is the purpose of that?

Reply:
It was to copy the structure of `withAuth` but agree it's not needed. Will remove.